Tests/scripts/hook_validations/structure.py

```diff
@@ -18,295 +29,281 @@
-            return True
+        file_id = loaded_file_data.get('id')
+        if not file_id:
+            # In integrations/scripts, the id is under 'commonfields'.
+            file_id = loaded_file_data.get('commonfields', {}).get('id')
+        if not file_id:
+            # In layout, the id is under 'layout'.
+            file_id = loaded_file_data.get('layout', {}).get('id')
+        return file_id
 
-        if not change_string:
-            change_string = run_command("git diff HEAD {0}".format(self.file_path))
+    def is_file_id_without_slashes(self):
+        # type: () -> bool
+        """Check if the ID of the file contains any slashes ('/').
 
-        is_added_from_version = re.search(r"\+([ ]+)?fromversion: .*", change_string)
-        is_added_from_version_secondary = re.search(r"\+([ ]+)?\"fromVersion\": .*", change_string)
+        Returns:
+            bool. Whether the file's ID contains slashes or not.
+        """
+        file_id = self.get_file_id_from_loaded_file_data(self.current_file)
+        if file_id and '/' in file_id:
+            self.is_valid = False
+            print_error(Errors.file_id_contains_slashes())
+            return False
+        return True
 
-        if is_added_from_version or is_added_from_version_secondary:
-            print_error("You've added fromversion to an existing file in the system, this is not allowed, please undo. "
-                        "the file was {}.".format(self.file_path))
-            self._is_valid = False
+    def is_id_modified(self, change_string=None):
+        # type: (Optional[str]) -> bool
+        """Check if the ID of the file has been changed.
 
-        return self._is_valid
+        Args:
+            change_string (string): the string that indicates the changes done on the file(git diff)
 
-    @staticmethod
-    def is_release_branch():
-        """Check if we are working on a release branch."""
-        diff_string_config_yml = run_command("git diff origin/master .circleci/config.yml")
-        if re.search(r'[+-][ ]+CONTENT_VERSION: ".*', diff_string_config_yml):
-            return True
+        Returns:
+            (bool): Whether the file's ID has been modified or not.
+        """
+        if self.is_renamed:
+            print_warning(Errors.id_might_changed())
+            return False
+
+        if not change_string:
+            change_string = run_command("git diff HEAD {}".format(self.file_path))
 
+        if re.search("[+-]( {2})?id: .*", change_string):
+            print_error(Errors.id_changed(self.file_path))
+            self.is_valid = False
+            return True
         return False
 
-    def _is_beta_integration(self):
-        """Checks if file is under Beta_integration dir"""
-        return re.match(BETA_INTEGRATION_REGEX, self.file_path, re.IGNORECASE) or \
-            re.match(BETA_INTEGRATION_YML_REGEX, self.file_path, re.IGNORECASE)
+    def is_valid_fromversion_on_modified(self, change_string=None):
```

Comment:
while we are refactoring everything here, change the verification to be based on `self._old_file` instead of `git diff`

should be part of the `BaseValidator`